### ISSUE TITLE
SC3028 for "${EUID:-$(id -u)}"

### ISSUE THREAD

**Issue by dereckson** [koalaman/shellcheck#2728]
- Rule Id (if any, e.g. SC1000): SC3028
- My shellcheck version (`shellcheck --version` or "online"): 0.9.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

if [ "${EUID:-$(id -u)}" -ne 0 ]; then
        echo "This command must be run as root." >&2
        exit 1
fi


```

#### Here's what shellcheck currently says:

```
if [ "${EUID:-$(id -u)}" -ne 0 ]; then
      ^---------------^ SC3028 (warning): In POSIX sh, EUID is undefined.

For more information:
  https://www.shellcheck.net/wiki/SC3028 -- In POSIX sh, EUID is undefined.
```

#### Here's what I wanted or expected to see:

Not a definitive idea, but the intent of the code is to use $EUID when it exists,
and fallback to id -u when it's not available, so I guess the warning is given
for a script having taking care of the fact EUID is undefined in POSIX.

I'm rather sceptical about code legibility with this kind of kludges and prefer
a straightforward `if [ "$(id -u)" -ne 0 ]` against a not-impacting gain of performance
to avoid to spawn an `id` process, but I'm not fully comfortable with the warning
in that context, so this report.
